src/tool_help.c

```diff
@@ -91,12 +91,21 @@ static void print_category(curlhelp_t category)
     if(len > longdesc)
       longdesc = len;
   }
-  if(longopt + longdesc > 80)
-    longopt = 80 - longdesc;
-
+  if(longopt + longdesc >= 80) {
+    longdesc -= 3;
+    longopt = 79 - longdesc;
+  }
   for(i = 0; helptext[i].opt; ++i)
     if(helptext[i].categories & category) {
-      printf(" %-*s %s\n", (int)longopt, helptext[i].opt, helptext[i].desc);
+      int opt = (int)longopt;
+      size_t desclen = strlen(helptext[i].desc);
+      if(opt + desclen >= 79) {
+        if(desclen < 79)
+          opt = 78 - (int)desclen;
+        else
+          opt = 0;
+      }
+      printf(" %-*s %s\n", opt, helptext[i].opt, helptext[i].desc);
```

Comment:
For some Windows consoles if the characters run to 80 then it makes the next line a blank line. for example here's what it looks like:
~~~
     --crlfile <file>                Certificate Revocation list
     --curves <list>                 (EC) TLS key exchange algorithms to request

 -d, --data <data>                   HTTP POST data
~~~

I think that the math needs to account for the spaces in the printf. But also I think there should be another space for the options that exceed the alignment and we end up with one space. here's what that looks like:

~~~
     --request-target <path>         Specify the target for this request
     --resolve <[+]host:port:addr[,addr]...> Resolve host+port to address
     --retry <num>                   Retry request if transient problems occur
~~~

cutting back a few columns:
~~~diff
diff --git a/src/tool_help.c b/src/tool_help.c
index 38a77f2..4e44f3a 100644
--- a/src/tool_help.c
+++ b/src/tool_help.c
@@ -91,21 +91,21 @@ static void print_category(curlhelp_t category)
     if(len > longdesc)
       longdesc = len;
   }
-  if(longopt + longdesc >= 80) {
+  if(longopt + longdesc >= 79) {
     longdesc -= 3;
-    longopt = 79 - longdesc;
+    longopt = 78 - longdesc;
   }
   for(i = 0; helptext[i].opt; ++i)
     if(helptext[i].categories & category) {
       int opt = (int)longopt;
       size_t desclen = strlen(helptext[i].desc);
-      if(opt + desclen >= 79) {
-        if(desclen < 79)
-          opt = 78 - (int)desclen;
+      if(opt + desclen >= 77) {
+        if(desclen < 77)
+          opt = 76 - (int)desclen;
         else
           opt = 0;
       }
-      printf(" %-*s %s\n", opt, helptext[i].opt, helptext[i].desc);
+      printf(" %-*s  %s\n", opt, helptext[i].opt, helptext[i].desc);
     }
 }
 
~~~

~~~
     --crlfile <file>                Certificate Revocation list
     --curves <list>                (EC) TLS key exchange algorithms to request
 -d, --data <data>                   HTTP POST data
~~~

~~~
     --request-target <path>         Specify the target for this request
     --resolve <[+]host:port:addr[,addr]...>  Resolve host+port to address
     --retry <num>                   Retry request if transient problems occur
~~~

Reply:
I think we could maybe save the fine tuning for #13171...